addons/calendar/models/calendar_event.py

```diff
@@ -355,6 +355,43 @@ def _compute_attendee(self):
             attendee = mapped_attendees[meeting.id]
             meeting.attendee_status = attendee.state if attendee else 'needsAction'
 
+    @api.model
+    def get_search_attendee_status(self, operator, value):
+        negative = operator in expression.NEGATIVE_TERM_OPERATORS
+        all_status = set([status[0] for status in Attendee.STATE_SELECTION])
+
+        # In case we have no value or non supported operator
+        if not value or operator not in ['=', '!=', 'in', 'not in']:
+            return all_status if negative else []
+
+        if operator == '=':
+            return {value}
+        elif operator == '!=':
+            return all_status - {value}
+        elif operator == 'in':
+            return value
+        elif operator == 'not in':
+            return all_status - set(value)
+
+    def _search_attendee_status(self, operator, value):
+        all_status = set([status[0] for status in Attendee.STATE_SELECTION])
+        search_status = self.get_search_attendee_status(operator, value)
+        if all_status == search_status:
+            return expression.TRUE_DOMAIN
+        elif not all_status:
+            return expression.FALSE_DOMAIN
+
+        self._cr.execute(
+            """
+              SELECT event_id
+                FROM calendar_attendee
+               WHERE state IN %s
+            """
+        , [tuple(search_status)])
+        searched_event_ids = tuple(event[0] for event in self._cr.fetchall())
+
+        return [('id', 'in', searched_event_ids)]
```

Comment:
Even better !
We could use `inselect` operator to avoid execute the query to get the event ids, we let the ORM doing it when it will parse the domain.

```suggestion
        query = self.env['calendar.attendee']._search([
	    ('state', 'in', list(search_status))
	])
        return [('id', 'inselect', query.select('event_id'))]
```